File: 민영/그리디 알고리즘/[5585]거스름돈.py
price = int(input())
count = 0

a = [500, 100, 50, 10, 5, 1]

change = 1000 - price

for i in range (6) :
    if change == 0: break
    elif change >= a[i] :
        count += change // a[i]
        change = change % a[i]
    else:
        continue

print(count)

File: 민영/그리디 알고리즘/[5585]거스름돈.cpp
#include <iostream>
using namespace std;

int main() {
    int a[6]= { 500, 100, 50, 10, 5, 1 };
    int price;
    cin >> price;
   
    int change = 1000 - price;
    int i = 0, count = 0;
    while (change != 0 && i != 6) {
        if (change >= a[i]) {
            count += change / a[i];
            //cout << "temp = " << change << ", a[ "<< i-1 <<" ] = "<< a[i - 1] << ", count = " << count << endl;
            change = change % a[i];
            i++;
        }
        else {
            i++;
        }
    }

    cout << count;
    return 0;
}